src/components/MovieListPage/MovieListPage.jsx

```diff
@@ -12,68 +13,122 @@ import "../MovieForm/style.css";
-      genres: ["Action & Adventure"],
-    },
+    { id: 1, name: "All" },
+    { id: 2, name: "Adventure" },
+    { id: 3, name: "Action" },
+    { id: 4, name: "Science Fiction" },
+    { id: 5, name: "Fantasy" },
+    { id: 6, name: "Family"},
+    { id: 7, name: "Comedy"}
   ]);
+  const [moviesInfo, setMoviesInfo] = useState([]);
+  // [
+  //   {
+  //     id: 1,
+  //     poster_path: "./images/Pulp fiction.jpg",
+  //     title: "Pulp fiction",
+  //     release_date: "1995",
+  //     vote_average: 7.2,
+  //     runtime: "2 hours 34 minutes",
+  //     overview:
+  //       "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.—Soumitra",
+  //     genres: ["Action & Adventure", "Criminal"],
+  //   },
+  //   {
+  //     id: 2,
+  //     poster_path: "./images/Inception.jpg",
+  //     title: "Inception",
+  //     release_date: "2011",
+  //     vote_average: 9.3,
+  //     runtime: "2 hours 46 minutes",
+  //     overview: "Sci-fi movie with Dicaprio",
+  //     genres: ["Action & Adventure"],
+  //   },
+  //   {
+  //     id: 3,
+  //     poster_path: "./images/Kill Bill.jpg",
+  //     title: "Kill Bill",
+  //     release_date: "1992",
+  //     vote_average: 7.1,
+  //     runtime: "1 hours 53 minutes",
+  //     overview: "Another criminal movie by Tarantino",
+  //     genres: ["Oscar winning Movie"],
+  //   },
+  //   {
+  //     id: 4,
+  //     poster_path: "./images/Avengers.jpg",
+  //     title: "Avengers",
+  //     release_date: "2019",
+  //     vote_average: 7.3,
+  //     runtime: "2 hours 26 minutes",
+  //     overview: "Movie by comics",
+  //     genres: ["Action & Adventure"],
+  //   },
+  // ]);
   const [selectedMovie, setSelectedMovie] = useState(null);
-  const [showModal, setShowModal] = useState(false);
+  const [isFound, setIsFound] = useState(false);
+  const [movieInfo, setMovieInfo] = useState({});
+
+  useEffect(() => {
+    console.log("Get all movies");
+    axios.get("http://localhost:4000/movies")
+    .then((response) => {
+      const movies = response.data.data;
+      setMoviesInfo(movies);
+    })
+    .catch((error) => {
+      console.log("Error while getting movies :" + error);
+    });
+  }, []);
+
+  useEffect(() => {
+    console.log("Search and filter movies");
+    axios.get("http://localhost:4000/movies")
+    .then((response) => {
+      const movies = response.data.data;
+      let filteredMovies = movies.filter(movie => movie.title.toLowerCase().includes(querySearch.trim().toLowerCase()))
```

Comment:
Currently, you try to filter and sort items on the FE side. At the same time, BE API allows filtering and sorting using request parameters(https://axios-http.com/docs/req_config).

Please, start to BE, and view http://localhost:4000/api-docs.



```
  axios.get("http://localhost:4000/movies", {
      params: {
        sortBy: sortOrder,
        search: querySearch,...
      }
    })
```
Please, use BE features